network-api/networkapi/wagtailpages/wagtail_hooks.py

```diff
@@ -167,6 +171,26 @@ def manage_index_pages_cache(request, page):
         parent.clear_index_page_cache(locale)
 
 
+@hooks.register("before_delete_page")
+def delete_historical_ab_tests_before_page_deletion(request, page):
```

Comment:
This function runs before any page is deleted. It was implemented to fix a bug I encountered where **any** page that had run an AB test before would be prevented from being deleted, due to a protected relationship between the page and the old `AbTest` model.

I have also created a bug in the wagtail-ab-testing repo [here](https://github.com/wagtail-nest/wagtail-ab-testing/issues/90) to make the team aware. I will post further details in that ticket soon!

Reply:
wagtail-ab-testing maintainer here 👋 

Great to see you found a workaround. Happy to review a PR that fixes this roadbump in ab-testing itself 😉